--- perfetto/sdk/perfetto.cc
+++ perfetto/sdk/perfetto.patched.cc
@@ -48972,6 +48972,14 @@
 
 // gen_amalgamated expanded: #include "perfetto/tracing/internal/track_event_interned_fields.h"
 
+void tr_expand_tag[[gnu::weak]](std::function<void(const std::string&)> set, const char *value) {
+    set(value);
+}
+namespace shynur::test_symbol {
+    void real_tr_expand_tag() {
+        ::tr_expand_tag([](const std::string&){}, "");
+    }
+}
 namespace perfetto {
 namespace internal {
 
@@ -49020,7 +49028,7 @@
     const char* value) {
   auto name = interned_data->add_debug_annotation_value_type_names();
   name->set_iid(iid);
-  name->set_name(value);
+  ::tr_expand_tag([name](const std::string& v) {name->set_name(v);}, value);
 }
 
 }  // namespace internal
@@ -61868,8 +61876,8 @@
 struct SequenceState {
   SequenceState(ProducerID, WriterID, ClientIdentity);
   ~SequenceState();
-  SequenceState(const SequenceState&) noexcept;
-  SequenceState& operator=(const SequenceState&) noexcept;
+  SequenceState(const SequenceState&);
+  SequenceState& operator=(const SequenceState&);
 
   ProducerID producer_id = 0;
   WriterID writer_id = 0;
@@ -62690,8 +62698,8 @@
       client_identity(c),
       chunks(/*initial_capacity=*/64) {}
 SequenceState::~SequenceState() = default;
-SequenceState::SequenceState(const SequenceState&) noexcept = default;
-SequenceState& SequenceState::operator=(const SequenceState&) noexcept =
+SequenceState::SequenceState(const SequenceState&) = default;
+SequenceState& SequenceState::operator=(const SequenceState&) =
     default;
 
 // +---------------------------------------------------------------------------+
